### PR TITLE
create jira issue if a new version is found

### PR DESCRIPTION
**Pull request by odra**
## What

* creates a new jira issue creation if source code changes are detected (branch job)
* Uses a `jiraUserPass`param as jenkins secret (username and password) but will use `integreatly-jira-bot` as secret name if there is no value for that param
* It looks for an open jira issue with the following labels before creating one: labels=`product-update` and `${product}` (product being the product name, i.e. 3scale)
* Creates a jira with the previous labels adding the pull request in ticket.

### CODE REVIEW

File: jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile [hunk 1]
```diff
@@ -108,6 +157,12 @@ node {
                 String body = "Created automatically by jenkins"
                 GHPullRequest pr = ghFindOrCreatePullRequest(gitHub.getRepository("${ghOwner}/${ghRepo}"), "${ghOwner}:${sourceBranch}", targetBranch, title, body)
                 println "Component Update PR = ${pr.getHtmlUrl()}"
+                def (found, key) = hasJiraUpdateIssue(credentials, productName)
+                if (!found) {
+                    createJiraIssue(jiraUserPass, pr.getHtmlUrl(), productName)
+                } else {
+                    println "[INFO] Issue already open for ${productName}: https://issues.jboss.org/browse/${key}"
+                }
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: Should we move this into it's own "Create Jira" stage, or is there a reason it needs to be in here?

Comment (mikenairn) [reply jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: Ah we use the pr object

File: jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile [hunk 2]
```diff
@@ -61,6 +61,60 @@ GHPullRequest ghFindOrCreatePullRequest(GHRepository repo, String head, String b
...
+
+def createJiraIssue(credentials, gitUrl, product, version) {
+    def body = """{
+        "fields": {
+            "project": {
+                "id": "12321620",
+                "key": "INTLY",
+                "name": "Integreatly"
+            },
+            "summary": "New version available for ${product}",
```
Comment (mikenairn) [inline jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: Can we add the version into here now? Maybe make the summary more like the ones that had been manually created "Update to Fuse 7.3" https://issues.jboss.org/browse/INTLY-1744

File: jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile (hunk 1 above)
Comment (odra) [reply jobs/delorean/jenkinsfiles/branch/ga/Jenkinsfile]: we only do it if a pr is going to be sent

**Pull request merged**